tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -233,6 +233,34 @@ def test_error_handling(scheduler, region, instance, pcluster_config_reader, clu
     )
 
 
+@pytest.mark.usefixtures("region", "os", "instance", "scheduler")
+@pytest.mark.slurm_protected_mode
+def test_slurm_protected_mode(
+    scheduler, region, pcluster_config_reader, clusters_factory, test_datadir, s3_bucket_factory
+):
+    """Test that slurm protected mode logic can handle bootstrap failure nodes."""
+    # Create S3 bucket for pre-install scripts
+    bucket_name = s3_bucket_factory()
+    bucket = boto3.resource("s3", region_name=region).Bucket(bucket_name)
+    bucket.upload_file(str(test_datadir / "preinstall.sh"), "scripts/preinstall.sh")
+    cluster_config = pcluster_config_reader(bucket=bucket_name)
+    cluster = clusters_factory(cluster_config)
+    remote_command_executor = RemoteCommandExecutor(cluster)
+    scheduler_commands = get_scheduler_commands(scheduler, remote_command_executor)
+    # Disable protected_mode by setting protected_failure_count to -1
+    _set_protected_failure_count(remote_command_executor, -1)
+    _inject_bootstrap_failures(cluster, bucket_name, pcluster_config_reader)
```

Comment:
Very minor: Would it make sense to move these functions into `_test_disable_protected_mode`? This might make the code more readable. Also I consider these functions as the "pre-req" for `_test_disable_protected_mode`, and if we move these into `_test_disable_protected_mode` we can move testing functions around without having to worry about the order of executions.

Similar comments apply to `_enable_protected_mode` and `_set_protected_failure_count` calls below

Reply:
Good idea! Done